tools/tertiary-analysis/seurat/seurat_dim_plot.xml

```diff
@@ -0,0 +1,90 @@
+<tool id="seurat_dim_plot" name="Plot dimension reduction" version="2.3.1">
+    <description>Graphs the output of a dimensional reduction technique (PCA by default). Cells are colored by their identity class.</description>
+    <macros>
+        <import>seurat_macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <expand macro="version" />
+    <command detect_errors="exit_code"><![CDATA[
+        seurat-dim-plot.R 
+        --input-object-file '$input' 
+        #if $reduction_use
+        --reduction-use '$reduction_use'
+        #end if
+        #if $dim_1
+        --dim-1 $dim_1
+        #end if
+        #if $dim_2
+        --dim-2 $dim_2
+        #end if
+        #if $cells_use
+        --cells-use $cells_use
+        #end if
+        #if $pt_size
+        --pt-size $pt_size
+        #end if
+        #if $label_size
+        --label-size $label_size
+        #end if
+        #if $do_label
+        --do-label $do_label
+        #end if
+        #if $group_by
+        --group-by $group_by
+        #end if
+        #if $plot_title
+        --plot-title '$plot_title'
+        #end if
+        --png-width $png_width
+        --png-height $png_height
+        --output-image-file $output_image_file
+    ]]></command>
+    <inputs>
+        <param type="data" name="input" format="rdata" label="RDS object" help='File in which a serialized R matrix object may be found'/>
+        <param type="text" name="reduction_use" value='pca' label="Dimensionality reduction" help='Which dimensionality reduction to use. Default is "pca", can also be "tsne", or "ica", assuming these are precomputed' />
+        <param type="integer" name="dim_1" value='1' label="X axis dimension" help='Dimension for x-axis' />
+        <param type="integer" name="dim_2" value='2' label="Y axis dimension" help='Dimension for y-axis' />
+        <param type="data" name="cells_use" format="tabular" label="Cells list" help='File to be used to derive a vector of cells to plot (default is all cells)' optional='true'/>
+        <param type="float" name="pt_size" value='1' label="Point size" help='Adjust point size for plotting' />
+        <param type="float" name="label_size" value='4' label="Label size" help='Sets size of labels'/>
+        <param type="boolean" name="do_label" checked="false" label='Whether to label the clusters' truevalue="TRUE" falsevalue="FALSE" />
+        <param type="text" name="group_by" value='ident' label='Group by' help='Group (color) cells in different ways' />
+        <param type="text" name="plot_title" format="text" label='Plot title' value='NULL' />
+        <param type="integer" name="png_width" value='1000' label='Width of png (px)' />
+        <param type="integer" name="png_height" value='1000' label='Height of png (px)'/>
+    </inputs>
+    <outputs>
+        <data name="output_image_file" format="png" from_work_dir="pca.png" />
```

Comment:
Perhaps add `label=` for the outputs here to help the user identify the output from the run history? Something like `"${tool.name} on ${on_string}: pca.png"` should suffice.